app/controllers/events_controller.rb

```diff
@@ -0,0 +1,73 @@
+class EventsController < ApplicationController
+  before_action :set_project
```

Comment:
use devise helper: before_action :authenticate_user! i guess it will simplyfy tour authorization logic